source/PluginDev/Assets/GooglePlayGames/Platforms/Android/AndroidTurnBasedMultiplayerClient.cs

```diff
@@ -25,70 +25,331 @@ public AndroidTurnBasedMultiplayerClient(AndroidJavaObject account)
+                            {
+                                OurUtils.Logger.e(string.Format("Could not find match {0}", matchId));
+                                callback(false, null);
+                            }
+                            else
+                            {
+                                callback(true, turnBasedMatch);
+                            }
+                        }
+                    }));
+                task.Call<AndroidJavaObject>("addOnFailureListener", new TaskOnFailedProxy(
+                    exception =>
+                    {
+                        callback(false, null);
+                    }
+                ));
+            }
         }
 
         public void AcceptFromInbox(Action<bool, TurnBasedMatch> callback)
         {
             callback = ToOnGameThread(callback);
-            // Task<Intent> getInboxIntent()
+
+            AndroidHelperFragment.ShowInboxUI((status, turnBasedMatch) =>
+            {
+                if (status != UIStatus.Valid)
+                {
+                    callback(false, null);
+                    return;
+                }
+
+                OurUtils.Logger.d("Passing converted match to user callback:" + turnBasedMatch);
+                callback(true, turnBasedMatch);
+            });
         }
 
         public void AcceptInvitation(string invitationId, Action<bool, TurnBasedMatch> callback)
         {
             callback = ToOnGameThread(callback);
-            // Task<TurnBasedMatch> acceptInvitation(@NonNull String invitationId)
+
+            FindInvitationWithId(invitationId, invitation =>
+            {
+                if (invitation == null)
+                {
+                    OurUtils.Logger.e("Could not find invitation with id " + invitationId);
+                    callback(false, null);
+                    return;
+                }
+
+                using (var task = mClient.Call<AndroidJavaObject>("acceptInvitation", invitationId))
+                {
+                    task.Call<AndroidJavaObject>("addOnSuccessListener", new TaskOnSuccessProxy<AndroidJavaObject>(
+                        turnBasedMatch =>
+                        {
+                            callback(true, AndroidJavaConverter.ToTurnBasedMatch(turnBasedMatch));
+                        }
+                    ));
+                    task.Call<AndroidJavaObject>("addOnFailureListener", new TaskOnFailedProxy(
+                        exception =>
+                        {
+                            callback(false, null);
+                        }
+                    ));
+                }
+            });
         }
 
         public void RegisterMatchDelegate(MatchDelegate del)
         {
-            // Task<Void> registerTurnBasedMatchUpdateCallback(@NonNull TurnBasedMatchUpdateCallback callback) 
-            // Task<Boolean> unregisterTurnBasedMatchUpdateCallback(@NonNull TurnBasedMatchUpdateCallback callback)
+            if (del != null)
```

Comment:
Because we have only RegisterMatchDelegate and don't have UnregisterMatchDelegate we should consider it is only one delegate is possible to set, so, null should unregister it. And any new should replace the old one.

Reply:
Similar to the one in AndroidClient, created an mMatchDelegate field and  replacing it every time this method is called.